src/boot/boot.c

```diff
@@ -2950,8 +2957,13 @@ static EFI_STATUS run(EFI_HANDLE image) {
                 (void) process_random_seed(root_dir);
 
                 err = image_start(image, entry);
-                if (err != EFI_SUCCESS)
-                        return err;
+                if (err != EFI_SUCCESS) {
+                        log_error_status(err, "Error starting image: %m");
+                        if (config.reboot_on_err)
+                                reboot_system();
```

Comment:
Rather than introducing the new knob and reboot system, maybe better to continue the loop on failure unconditionally.